### PR TITLE
Makefile cleanup sizes and travis-ci s3 deploy

### PR DESCRIPTION
**Pull request by dagar**
- pulls more of the travis-ci s3 deploy into the repo so we can potentially migrate to another CI system
 - fixed the sizes output and added verbose compiler version to cmake (#6322)
 - fixed filenames for firmware uploaded to s3 (was broken by the changes yesterday)
 - fixed some broken git version display in cmake
 - Makefile organization
 - simplified .travis.yml
 - added a print to know which config the nuttx patch was being applied to
 - docker_run.sh now respects PX4_DOCKER_REPO for setting the docker image, but defaults to the good production nuttx image

### CODE REVIEW

**Comment by dagar** [comment]
@davids5 Full compiler --version is now printed in cmake configure.

	-- PX4 VERSION: v1.5.2-633-gd9b9aa6
	-- CONFIG: nuttx-auav-x21-default
	-- CMAKE_MODULE_PATH: /home/runner/Firmware/cmake
	-- Nuttx build for auav-x21 on m4 hardware, using nsh with ROMFS on px4fmu_common
	-- The ASM compiler identification is GNU
	-- Found assembler: /usr/lib/ccache/arm-none-eabi-gcc
	-- Found PythonInterp: /usr/bin/python (found version "2.7.12") 
	-- C compiler: arm-none-eabi-gcc (GNU Tools for ARM Embedded Processors) 5.4.1 20160919 (release) [ARM/embedded-5-branch revision 240496]
	-- C++ compiler: arm-none-eabi-g++ (GNU Tools for ARM Embedded Processors) 5.4.1 20160919 (release) [ARM/embedded-5-branch revision 240496]

File: Makefile [hunk 1]
```diff
@@ -151,56 +150,20 @@ posix: posix_sitl_default
 broadcast: posix_sitl_broadcast
 
 # Multi- config targets.
-
 eagle_default: posix_eagle_default qurt_eagle_default
 eagle_legacy_default: posix_eagle_legacy qurt_eagle_legacy
 excelsior_default: posix_excelsior_default qurt_excelsior_default
 
-# Deprecated config targets.
```
Comment (jgoppert) [inline Makefile]: I assume this was intentional? It is pretty outdated now.

**Comment by davids5** [comment]
@dagar - Perfect!

File: Makefile (hunk 1 above)
Comment (dagar) [inline Makefile]: Yes, we've had that message for a long time.

**Pull request merged**